include/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster.hpp

```diff
@@ -43,6 +47,19 @@ class Thruster : public Dynamics
         const String& aName = "Thruster"
     );
 
+    /// @brief Constructor
+    ///
+    /// @param aSatelliteSystem A satellite system
+    /// @param aGuidanceLaw A guidance law
+    /// @param applicabilityInterval An applicability interval
+    /// @param aName A name
+    Thruster(
+        const SatelliteSystem& aSatelliteSystem,
+        const Shared<const GuidanceLaw>& aGuidanceLaw,
+        const Interval& applicabilityInterval,
```

Comment:
Hmmmmmmm, I don't think the thruster should have a concept of an interval, that IMO should be defined at the guidance law level.